apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.app;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.inject.spi.CDI;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.apollocurrency.aplwallet.api.dto.DurableTaskInfo;
+import com.apollocurrency.aplwallet.apl.core.account.Account;
+import com.apollocurrency.aplwallet.apl.core.chainid.BlockchainConfig;
+import com.apollocurrency.aplwallet.apl.core.chainid.BlockchainConfigUpdater;
+import com.apollocurrency.aplwallet.apl.core.db.DatabaseManager;
+import com.apollocurrency.aplwallet.apl.core.db.TransactionalDataSource;
+import com.apollocurrency.aplwallet.apl.core.db.cdi.Transactional;
+import com.apollocurrency.aplwallet.apl.crypto.Convert;
+import com.apollocurrency.aplwallet.apl.crypto.Crypto;
+import com.apollocurrency.aplwallet.apl.util.Constants;
+import com.apollocurrency.aplwallet.apl.util.env.dirprovider.ConfigDirProvider;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class GenesisImporter {
+
+    private byte[] CREATOR_PUBLIC_KEY;
+    public static long CREATOR_ID;
+    public static long EPOCH_BEGINNING;
+    public static final String LOADING_STRING_PUB_KEYS = "Loading public keys %d / %d...";
+    public static final String LOADING_STRING_GENESIS_BALANCE = "Loading genesis amounts %d / %d...";
+
+    private BlockchainConfig blockchainConfig;
+    private ConfigDirProvider configDirProvider;
+    private AplAppStatus aplAppStatus;
+
+    private BlockchainConfigUpdater blockchainConfigUpdater;
+    private DatabaseManager databaseManager;
+    private String genesisTaskId;
+    private JsonNode genesisAccountsJSON = null;
+    private List<String> publicKeys;
+    private Map<String, Long> balances;
+    private byte[] computedDigest;
+    private ObjectMapper mapper = new ObjectMapper();
+    private String genesisParametersLocation;
+
+    @Inject
+    public GenesisImporter(BlockchainConfig blockchainConfig, ConfigDirProvider configDirProvider,
+                           BlockchainConfigUpdater blockchainConfigUpdater,
+                           DatabaseManager databaseManager,
+                           AplAppStatus aplAppStatus,
+                           GenesisImporterProducer genesisImporterProducer) {
```

Comment:
Maybe just inject String with named qualifier?

Reply:
I tried, but in such case it's harder to write unit test with 'injected String'.